### PR TITLE
Pool 1789 reserve does not correctly implement

### PR DESCRIPTION
**Pull request by asselstine**

### CODE REVIEW

File: contracts/Reserve.sol [hunk 1]
```diff
@@ -8,6 +8,9 @@ import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 
 import "./interfaces/IReserve.sol";
 import "./libraries/ObservationLib.sol";
+import "./libraries/RingBufferLib.sol";
+
+import "hardhat/console.sol";
```
Comment (PierrickGT) [inline contracts/Reserve.sol:13]: ✂️

File: contracts/Reserve.sol [hunk 2]
```diff
@@ -225,12 +237,28 @@ contract Reserve is IReserve, Manageable {
         }
     }
 
+    function _getOldestObservation(uint24 _nextIndex)
```
Comment (PierrickGT) [inline contracts/Reserve.sol:240]: Add Natspec doc

File: test/Reserve.test.ts [hunk 3]
```diff
@@ -254,19 +254,6 @@ describe('Reserve', () => {
         });
     });
 
-    describe('_getReserveAccumulatedAt()', () => {
```
Comment (PierrickGT) [inline test/Reserve.test.ts:257]: We should still test this function for edge cases. That's why coverage has decreased.

**Review by PierrickGT — CHANGES_REQUESTED** [review]
